### PR TITLE
User management UX fixes

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/2296

### Code Changes

* [x] Clean up admin UI README
* [x] Fix button stylings so when you hover they don't look unreadable
* [x] Refactor DeleteUserModal to use formik which allows us to do better validation
* [x] Refactor DeleteUserModal such that it isn't living inside the menu (which prevented tabbing from working)
* [x] Use common search bar component for user management page
* [x] Add cypress tests for deleting a user

### Steps to Confirm

* [ ] Poke around user management to make sure things look better
* [ ] Delete a user

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

Deleting a user tabs properly now ☺️ and has better validation

https://user-images.githubusercontent.com/24641006/217605978-82eca9a5-8571-4b55-9a76-3bd576fc4743.mov

### CODE REVIEW

File: clients/admin-ui/src/features/user-management/UserManagementRow.tsx [hunk 1]
```diff
@@ -24,46 +25,67 @@ interface UserManagementRowProps {
 
 const UserManagementRow: React.FC<UserManagementRowProps> = ({ user }) => {
   const router = useRouter();
+  const modal = useDisclosure();
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/user-management/UserManagementRow.tsx:28]: Super nit here 😬 Could this be renamed to `deleteModal` instead? It'll make it obvious what modal it goes too and if we add more modals in the future this one won't accidentally get reused.

**Review by TheAndrewJackson — APPROVED** [review]
Great work! Just one small nit. I'm happy that we're slowly converting everything over to forkmik 

scheming-evil-plan

File: clients/admin-ui/src/features/user-management/UserManagementRow.tsx (hunk 1 above)
Comment (allisonking) [reply clients/admin-ui/src/features/user-management/UserManagementRow.tsx:28]: heh you got it!

**Comment by allisonking** [comment]
privacy center check passed locally. the CI test seems to be failing across multiple branches, will need to look into it separately...

**Pull request merged**